src/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -19,76 +19,116 @@ APPLICATION_INFO::~APPLICATION_INFO()
 }
 
 HRESULT
-APPLICATION_INFO::GetOrCreateApplication(
+APPLICATION_INFO::CreateHandler(
     IHttpContext *pHttpContext,
-    std::unique_ptr<IAPPLICATION, IAPPLICATION_DELETER>& pApplication
+    std::unique_ptr<IREQUEST_HANDLER, IREQUEST_HANDLER_DELETER>& pHandler
 )
 {
     HRESULT             hr = S_OK;
 
-    SRWExclusiveLock lock(m_applicationLock);
-
-    auto& httpApplication = *pHttpContext->GetApplication();
-
-    if (m_pApplication != nullptr)
     {
-        if (m_pApplication->QueryStatus() == RECYCLED)
-        {
-            LOG_INFO(L"Application went offline");
+        SRWSharedLock lock(m_applicationLock);
+        
+        RETURN_IF_FAILED(hr = TryCreateHandler(pHttpContext, pHandler));
 
-            // Call to wait for application to complete stopping
-            m_pApplication->Stop(/* fServerInitiated */ false);
-            m_pApplication = nullptr;
-            m_pApplicationFactory = nullptr;
+        if (hr == S_OK)
+        {
+            return S_OK;
         }
-        else
+    }
+    
+    {
+        SRWExclusiveLock lock(m_applicationLock);
+        
+        // check if other thread created application
+        RETURN_IF_FAILED(hr = TryCreateHandler(pHttpContext, pHandler));
```

Comment:
This section is a bit vague, but I think I get it. Can you expand/elaborate about the order of these calls as comments?